core/iwasm/libraries/lib-wasi-threads/test/spawn_stress_test.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2023 Amazon.com Inc. or its affiliates. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#ifndef __wasi__
+#error This example only compiles to WASM/WASI target
```

Comment:
are we using any specific WASM/WASI instruction here?

Reply:
looks like we're not but I'm not sure we should allow anyone running this test without WASI define, should we? 

I mean, I checked this code only with WASI sdk, where this is defined. Without it, you won't be able to run `build.sh` etc